[FIX] hr_attendance: review menu layout

chgo-odoo opened the pull request:
This PR fixes the layout of the hr attendance menu.

Prior to this commit, the layout was a bit cluttered and had a lack of
white space, making it difficult for the user to extract the information
he needed.

To fix these issues, we adapt the design of the menu, mainly by
reorganizing the layout and adding more white space.

With this commit, we were also able to clean the files, reducing the
number of lines needed to achieve a good visual result.

task-3577098

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/140454).

stefanorigano commented on addons/hr_attendance/static/src/components/attendance_menu/attendance_menu.xml:
```diff
@@ -3,58 +3,35 @@
     <t t-if="this.state.isDisplayed">
-        <Dropdown position="'bottom-end'" beforeOpen.bind="searchReadEmployee" menuClass="`p-0 ${ ui.isSmall ? 'start-0 w-100 mh-100 bottom-0 d-flex flex-column' : ''}`">
-        <div class="d-flex flex-column flex-grow-1">
+        <Dropdown position="'bottom-end'" beforeOpen.bind="searchReadEmployee" menuClass="`p-2 ${ ui.isSmall ? 'd-flex flex-column' : ''}`">
```
Adding `d-flex flex-column` on mobile look unnecessary... leftovers from a prev. implementation?

stefanorigano replied:
I'd suggest adding a `pb-3` 😉

stefanorigano commented on addons/hr_attendance/static/src/components/attendance_menu/attendance_menu.scss:
```diff
@@ -1,32 +1,3 @@
-.o_hr_attendance_systray_check_in_btn{
-  width: 150px;
-  height: 40px;
+    font-variant-numeric: tabular-nums;
```
Note for the posterity: one day we should create an utility class for this 🤔

stefanorigano reviewed (changes requested):
Hello @chgo-odoo , very good, thanks 👍 

Some remarks:
- Adapt the PR/commit description since these changes affects desktop as well (not just mobile
- There's a conflict, you should rebase 😉
- [FP] Invert "Since" and "Before" position, it's more logical in chronological order 

| This PR. | Proposal |
|--------|--------|
| ![image](https://github.com/odoo/odoo/assets/8558351/f9edbcf0-095e-478d-a2c9-8552c570ac67) | ![image](https://github.com/odoo/odoo/assets/8558351/0ca83c45-7902-4b0e-b4fc-15fe2d11c2bc) | 

- [FP] **IF** all entries are visible ("Since", "Before" and "total today") , render the button inside the grey box.  

| This PR. | Proposal |
|--------|--------|
| ![image](https://github.com/odoo/odoo/assets/8558351/f9edbcf0-095e-478d-a2c9-8552c570ac67) | ![image](https://github.com/odoo/odoo/assets/8558351/420b9f8a-11f4-4588-bcdf-734fe8511c87) | 

...the rest in the code, thanks!

chgo-odoo commented:
Hi @stefanorigano, thanks for the detailed review !

I've made the requested changes, if you could have a look at it?

Thanks !

stefanorigano reviewed (approved):
Perfect, thanks @chgo-odoo 👍

stefanorigano commented:
@odoo/rd-hr , FYI, design approved by FP.
Could you have a look  and eventually r+ ?
Thanks!

sofiagvaladze commented:
@chgo-odoo thanks for your work :raised_hands: 

Can you delete empty enterprise branch and after that rebase and push community branch? 

Once it's green, I will merge it :rocket:

chgo-odoo commented:
Hi @sofiagvaladze thanks for taking care of this one 👋

I just deleted the empty enterprise branch and rebased community as requested but runbot keeps building that empty branch 🫣

sofiagvaladze commented:
@chgo-odoo it seems good now :crossed_fingers: thanks!

sofiagvaladze commented:
@chgo-odoo unrelated error is failing.. the thing is, even if I were to r+ it, it won't be merged unless runbot is green.

Rebase and push again in a bit and tag me once the branch is ready :rocket:

chgo-odoo commented:
Thanks @sofiagvaladze !

Runbot is (finally) green, can we move this forward? 🚀

Cheers !

sofiagvaladze commented:
robodoo r+
